miri-script/src/commands.rs

```diff
@@ -0,0 +1,711 @@
+                self.rustc_push(&github_user, &branch),
+        }
+    }
+    fn find_sysroot(&mut self) {
+        let current_sysroot = std::env::var("MIRI_SYSROOT").unwrap_or_default();
+
+        if !current_sysroot.is_empty() {
+            // Sysroot already set, use that.
+            let current_value =
+                self.extra_flags.insert("MIRI_SYSROOT".to_string(), current_sysroot.clone());
+            assert!(current_value.is_none() || current_value.unwrap() == current_sysroot);
+            return;
+        }
+        let miri_test_target = std::env::var("MIRI_TEST_TARGET").unwrap_or_default();
+        let target = (!miri_test_target.is_empty()).then_some(miri_test_target.as_str());
+        let sysroot = self.build_sysroot(target);
+        self.extra_flags.insert("MIRI_SYSROOT".into(), sysroot);
+    }
+
+    fn build_sysroot(&self, target: Option<&str>) -> String {
+        let manifest_path = path!(self.miri_dir / "cargo-miri" / "Cargo.toml");
+        let manifest_path = manifest_path.to_string_lossy().to_string();
+        let active_toolchain = self.get_toolchain_override();
+        let mut args = [
+            &active_toolchain,
+            &self.cargo_extra_flags,
+            "--manifest-path",
+            &manifest_path,
+            "--",
+            "miri",
+            "setup",
+        ]
+        .to_vec();
+        if let Some(target) = target {
+            args.extend(["--target", target]);
+        }
+        args.push("--print-sysroot");
+        let output = cmd("cargo", &args).read();
+        if output.is_err() {
+            // Run it again so the user can see the error.
+            // This time without printing a sysroot though.
+            args.pop();
+            let _ = cmd("cargo", &args).run().expect("`cargo miri setup` failed");
+        }
+
+        output.unwrap()
+    }
+}
+
+impl MiriRunner<'_> {
+    fn bench(&self, benches: &[String]) -> Result<()> {
+        // Make sure we have an up-to-date Miri installed
+        let benches_dir = path!(self.miri_dir / "bench-cargo-miri");
+        let benches = if benches.is_empty() {
+            std::fs::read_dir(&benches_dir)?
+                .filter_map(|path| {
+                    path.ok()
+                        .filter(|dir| dir.file_type().map(|t| t.is_dir()).unwrap_or(false))
+                        .map(|p| p.file_name().to_string_lossy().into_owned())
+                })
+                .collect()
+        } else {
+            benches.to_owned()
+        };
+        for bench in benches {
+            let current_bench_dir = path!(benches_dir / bench / "Cargo.toml");
+            let current_bench_dir = current_bench_dir.to_string_lossy().into_owned();
+            let toolchain = self.get_toolchain_override();
+            let command =
+                format!("cargo {} miri run --manifest-path {current_bench_dir}", toolchain);
+            cmd("hyperfine", ["-w", "1", "-m", "5", "--shell=none", &command]).run()?;
```

Comment:
Note that this changed in the shell script recently, so the rust version needs updating